src/Sarif.UnitTests/FileRegionsCacheTests.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.ObjectModel;
+using FluentAssertions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.UnitTests
+{
+    public class FileRegionsCacheTests
+    {
+        //                                 0          10          20         30
+        //                                 01234 5 6789012 3 45678901 2 345679012
+        private const string s_testText = "line1\r\n line2\r\n  line3\r\n   line4";
+
+        private static ReadOnlyCollection<Tuple<string, Region, Region>> s_testCases =
+            new ReadOnlyCollection<Tuple<string, Region, Region>>(new Tuple<string, Region, Region>[]
+            {
+                // Regions specified only by start line
+                new Tuple<string, Region, Region>(
+                    "line1",
+                    new Region() { StartLine = 1 }, 
+                    new Region() { StartLine = 1, EndLine = 1, StartColumn = 1, EndColumn = 5, CharOffset = 0, CharLength = 5 })
+            });
+
+
+        [Fact]
```

Comment:
Rather than looping on the test cases yourself, you could use xUnit's [`Theory` and `MemberData` attributes](https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/).

Reply:
you can't do this defining a strong typed test data class. i have taken this suggestion, but the result is not to use ClassData/MemberData/TheoryData. maybe there's a new way to mix these paradigms I haven't found? 

---
In reply to: [202753131](https://github.com/Microsoft/sarif-sdk/pull/945#discussion_r202753131) [](ancestors = 202753131)

Reply:
(Replied in email. You are right.)